src/DistrictResources/DistrictResources.cpp

```diff
@@ -1,4 +1,277 @@
+        }
+    }
+
+    FireStation DistrictResources::findFireStation(uint32_t fireStationID) {
+        for (auto station : districtFireStations)
+        {
+            if (station.getFireStationID() == fireStationID)
+            {
+                return station;
+            }
+        }
+        return {};
+    }
+
+    void DistrictResources::processVehicleSampleData(const std::string& fileName)
+    {
+        std::vector<std::vector<std::string>> vehicles;
+        std::vector<std::string> row;
+        std::string line, word;
+
+        std::fstream file (fileName, std::ios::in);
+        while(getline(file, line))
+        {
+            row.clear();
+
+            std::stringstream str(line);
+
+            while(getline(str, word, ','))
+                row.push_back(word);
+            vehicles.push_back(row);
+        }
+
+        bool header = true;
+
+        for (std::vector<std::string> entry : vehicles)
+        {
+            if (header)
+            {
+                header = false;
+                continue;
+            }
+            std::string vehicleType = entry.at(1);
+            Vehicle newVehicle{};
+            if (vehicleType == "Engine")
+            {
+                newVehicle = *new FireEngine();
+            }
+            else if (vehicleType == "Ladder")
+            {
+                newVehicle = *new FireLadder();
+            }
+
+            newVehicle.setVehicleID(stoi(entry.at(0)));
+            newVehicle.setHomeFireStation(findFireStation(stoi(entry.at(2))));
+            newVehicle.setMaxNumCrew(stoi(entry.at(3)));
+            newVehicle.setCurNumCrew(stoi(entry.at(4)));
+            newVehicle.setMaxWaterVolumeLitres(stof(entry.at(5)));
+            newVehicle.setCurWaterVolumeLitres(stof(entry.at(6)));
+            newVehicle.setWorkFactor(stof(entry.at(7)));
+
+            std::cout << "Adding vehicle #" << newVehicle.getVehicleID() << " to global list" <<std::endl;
+            districtVehicles.push_back(newVehicle);
+        }
+    }
+
+    void DistrictResources::processFireStationSampleData(const std::string &fileName)
+    {
+        std::vector<std::vector<std::string>> stations;
+        std::vector<std::string> row;
+        std::string line, word;
+
+        std::fstream file (fileName, std::ios::in);
+        while(getline(file, line))
+        {
+            row.clear();
+
+            std::stringstream str(line);
+
+            while(getline(str, word, ','))
+                row.push_back(word);
+            stations.push_back(row);
+        }
+
+        bool header = true;
+
+        for (std::vector<std::string> entry : stations)
+        {
+            if (header)
+            {
+                header = false;
+                continue;
+            }
+
+            FireStation newStation{};
+
+            auto newLocation = new Location();
+            newLocation->setXCoord(stof(entry.at(1)));
+            newLocation->setYCoord(stof(entry.at(2)));
```

Comment:
I suggest using constructors with parameters to create objects an initialisation lists, because it reduces the amount of error-handling code that you need. If any of the parameters fail (e.g., an `stof()`), then the `Location` object will fail to construct. With this code, if you caught an exception on `stof()`, you would probably have a memory leak on `newLocation`. See, for example, https://stackoverflow.com/a/8523361/2769271